### PR TITLE
feat: Add Broker JWT sync

### PR DESCRIPTION
**Pull request by GraceRuan**
Add Broker JWT sync

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
     "express-session": "^1.18.0",
     "file-stream-rotator": "^1.0.0",
     "helmet": "^7.1.0",
+    "libsodium-wrappers": "^0.7.15",
```
Comment (mbystedt) [inline package.json:47]: Unnecessary package as it duplicates the crypto built-in library

File: src/github/github.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      controllers: [GithubController],
+    }).compile();
+
+    controller = module.get<GithubController>(GithubController);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+});
```
Comment (mbystedt) [inline src/github/github.controller.spec.ts:18]: Remove file

File: src/github/github.controller.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  @Post('update-secret')
+  async updateSecret(
+    @Body('repoUrl') repoUrl: string,
+    @Body('secretName') secretName: string,
+    @Body('secretValue') secretValue: string,
+  ): Promise<void> {
+    await this.githubService.updateSecret(repoUrl, secretName, secretValue);
+  }
+}
```
Comment (mbystedt) [inline src/github/github.controller.ts:16]: Remove file

File: src/github/github.service.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    // Subscribe to the Redis channel
+    this.redisService.subscribeAndProcess(REDIS_PUBSUB.VAULT_SERVICE_TOKEN, async (event) => {
+      try {
+        const { clientId, environment, project, service, scmUrl } = event.data as {
+          clientId: string;
+          environment: string;
+          project: string;
+          service: string;
+          scmUrl: string;
+        };
```
Comment (mbystedt) [inline src/github/github.service.ts:42]: Ignore event if environment is not tools.

File: src/github/github.service.ts [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        const path = `${environment}/${project}/${service}`;
+        const kvData= await this.vaultService.getKv(VAULT_KV_APPS_MOUNT,path);
+        const value = kvData[`broker-jwt:${clientId}`];
```
Comment (mbystedt) [inline src/github/github.service.ts:45]: Loop over all keys and sync all values.

File: src/github/github.service.ts [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        if (!value) {
+          console.error(`No value found for client_id: ${clientId}`);
+          return;
+        }
+        if (scmUrl) {
```
Comment (mbystedt) [inline src/github/github.service.ts:50]: Check if scmUrl is GitHub. Bonus points if it checks the org and if the GitHub app has access.

File: src/github/github.service.ts [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+          const secretName = 'BROKER_JWT';
```
Comment (mbystedt) [inline src/github/github.service.ts:51]: Keep same name from vault.

File: src/github/github.service.ts [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+      return { owner: match[1], repo: match[2] };
+    }
+    throw new Error('Invalid GitHub URL');
+  }
+
+  // Implement encryption logic
+  private async encryptSecret(secretValue: string, publicKey: string): Promise<string> {
+
+    try {
+      await sodium.ready;
```
Comment (mbystedt) [inline src/github/github.service.ts:124]: convert to builtin crypto library. Hint! chagtpt is your friend:
rewrite using nodejs api: const encryptedUint8Array = sodium.crypto_box_seal(secretUint8Array, publicKeyUint8Array);

File: src/vault/vault.service.ts [hunk 9]
```diff
@@ -33,6 +33,25 @@ export class VaultService {
...
+  public async getKv(mount: string, path: string): Promise<any> {
+    try {
+      const config = this.prepareConfig();
+      config.headers['Content-Type'] = 'application/json';
+      const observable = this.httpService.get(`${this.vaultAddr}/v1/${mount}/data/${path}`, config);
+      const response: AxiosResponse<any> = await lastValueFrom(observable);
+      const kvData = response.data?.data?.data;
+
+      if (!kvData) {
+        throw new Error('No data found at the specified path');
```
Comment (mbystedt) [inline src/vault/vault.service.ts:45]: throw more specific error

File: src/vault/vault.service.ts [hunk 10]
```diff
@@ -33,6 +33,25 @@ export class VaultService {
...
+      }
+
+      return kvData;
+    } catch (error) {
```
Comment (mbystedt) [inline src/vault/vault.service.ts:49]: errors should only be caught by caller. Unnecessary try/catch

**Review by mbystedt — CHANGES_REQUESTED** [review]

**Comment by GraceRuan** [comment]
Will be created in a new branch

**Pull request closed without merging**